### PR TITLE
[msal-browser] Update request initialization code

### PR DESCRIPTION
**Pull request by pkanher617**
This PR updates the PublicClientApplication class in msal-browser to do the following:

- Rename initializer methods to be more descriptive
- Update acquireTokenSilent() to initialize the request authority and correlation ID
- Update all APIs to set default scopes

This PR also adds the following behavior that was unfinished:
- Add correlationID to all request flows.

### CODE REVIEW

File: lib/msal-browser/src/app/PublicClientApplication.ts [hunk 1]
```diff
@@ -661,11 +689,23 @@ export class PublicClientApplication {
...
-    private generateLogoutRequest(logoutRequest?: EndSessionRequest): EndSessionRequest {
-        return {
-            ...logoutRequest,
-            postLogoutRedirectUri: this.getPostLogoutRedirectUri(logoutRequest ? logoutRequest.postLogoutRedirectUri : "")
+    /**
+     * Initializer for the logout request.
+     * @param logoutRequest 
+     */
+    private initializeLogoutRequest(logoutRequest?: EndSessionRequest): EndSessionRequest {
+        const validLogoutRequest = {
```
Comment (DarylThayil) [inline lib/msal-browser/src/app/PublicClientApplication.ts:697]: what if logout request is null, will `...logoutRequest` throw?

Comment (DarylThayil) [reply lib/msal-browser/src/app/PublicClientApplication.ts:697]: @jasonnutter  cc

Comment (DarylThayil) [reply lib/msal-browser/src/app/PublicClientApplication.ts:697]: I think it will throw

Comment (jmckennon) [reply lib/msal-browser/src/app/PublicClientApplication.ts:697]: yeah, just tested it, spread operator can't be called on null

Comment (jasonnutter) [reply lib/msal-browser/src/app/PublicClientApplication.ts:697]: Spreading `null` should work inside objects (but not arrays).

**Pull request merged**